plugin/storage/es/spanstore/reader.go

```diff
@@ -135,24 +136,51 @@ type timeRangeIndexFn func(indexName string, startTime time.Time, endTime time.T
 
 type sourceFn func(query elastic.Query, nextTime uint64) *elastic.SearchSource
 
-func getTimeRangeIndexFn(archive, useReadWriteAliases bool) timeRangeIndexFn {
+func getTimeRangeIndexFn(archive, useReadWriteAliases bool, remoteReadClusters []string) timeRangeIndexFn {
 	if archive {
 		var archivePrefix string
 		if useReadWriteAliases {
 			archivePrefix = archiveReadIndexSuffix
 		} else {
 			archivePrefix = archiveIndexSuffix
 		}
-		return func(indexName string, startTime time.Time, endTime time.Time) []string {
+		return addRemoteReadClusters(func(indexName string, startTime time.Time, endTime time.Time) []string {
 			return []string{archiveIndex(indexName, archivePrefix)}
-		}
+		}, remoteReadClusters)
 	}
+
 	if useReadWriteAliases {
-		return func(indices string, startTime time.Time, endTime time.Time) []string {
+		return addRemoteReadClusters(func(indices string, startTime time.Time, endTime time.Time) []string {
 			return []string{indices + "read"}
+		}, remoteReadClusters)
+	}
+
+	return addRemoteReadClusters(timeRangeIndices, remoteReadClusters)
+}
+
+// Elasticsearch cross cluster query api example: GET /twitter,cluster_one:twitter,cluster_two:twitter/_search
+// Add a remote cluster prefix for each cluster and for each index.
+// The special remote cluster "local" leaves the prefix off of it's subset of indexes resulting in a standard local query
+func addRemoteReadClusters(fn timeRangeIndexFn, remoteReadClusters []string) timeRangeIndexFn {
+	return func(indices string, startTime time.Time, endTime time.Time) []string {
+		jaegerIndices := fn(indices, startTime, endTime)
+		if len(remoteReadClusters) > 0 {
+			remoteJaegerIndices := []string{}
+			for _, remoteCluster := range remoteReadClusters {
+				for _, jaegerIndex := range jaegerIndices {
+
+					if remoteCluster != "local" {
```

Comment:
Can we avoid this? We can require not adding it via flags.

Reply:
Some options:
 --es.remote-read-clusters-remote-only
--es.remote-read-clusters-no-local

What do you think?

Reply:
I might be missing something, but why to flags? The names on the flag will be added to indices. If local is added it will be added. if it is not added it won't be added.  I just want to avoid unnecessary checks in the reader.

Reply:
Ahh, I misunderstood. Are you suggesting we always keep the local un-prefixed index by default? I really struggled to figure out how we should toggle on and off the local cluster reference. I've seen use cases where you have an elastic query node that only has remote clusters and no data local.

Current:
  Input: --es.remote-read-clusters=local,cluster_one
  Output: /twitter,cluster_one:twitter
Alternate?:
 Input: --es.remote-read-clusters=cluster_one
 Output: /twitter,cluster_one:twitter

Reply:
>  I've seen use cases where you have an elastic query node that only has remote clusters and no data local.

can you provide any references to this use-case? 

Then I guess we cannot go with the following as it will always add local data.
```
Alternate?:
Input: --es.remote-read-clusters=cluster_one
Output: /twitter,cluster_one:twitter
```

Reply:
Thanks for the input! This does make things much simpler! The no local use case can be addressed in the future if anyone ever needs it.